maven-embedder/src/main/java/org/apache/maven/cli/props/PropertiesLoader.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.cli.props;
+
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.util.Enumeration;
+import java.util.StringTokenizer;
+
+import static org.apache.maven.cli.props.InterpolationHelper.substVars;
+
+public class PropertiesLoader {
+
+    public static final String INCLUDES_PROPERTY = "${includes}"; // mandatory includes
+
+    public static final String OPTIONALS_PROPERTY = "${optionals}"; // optionals include
```

Comment:
Maybe we can make optionals more succint with `${includes?}` like we have in here: https://maven.apache.org/plugins/maven-assembly-plugin/faq.html#dashClassifier

Reply:
This is the property key, you want to rename and use `${includes?} = ${user.home}/.m2/maven.properties` ?

Reply:
Yes, as an idea. Maybe it is a bad one. I am open to ideas. Comments should read "mandatory includes"/"optional includes", but the property names aren't consistent in this regard.

Reply:
Another possibility is to have a single list, and if the last character is a `?`, then make it optional.

`${includes}` = ${user.home}/.m2/maven.properties?, ${session.rootDirectory}/.mvn/maven.properties?`